File: Leetcode_905.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        nums2 = []  # define nums2 inside the function
        for i in range(len(nums)):
            if nums[i] % 2 == 0:
                nums2.append(nums[i])
        for j in range(len(nums)):
            if nums[j] % 2!=0 :
                nums2.append(nums[j])
        return nums2  # return the result

File: Leetcode_905.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int[] result = new int[nums.length];
        int index = 0;

        // First add evens
        for (int num : nums) {
            if (num % 2 == 0) {
                result[index++] = num;
            }
        }

        // Then add odds
        for (int num : nums) {
            if (num % 2 != 0) {
                result[index++] = num;
            }
        }

        return result;
    }
}
